src/cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -373,6 +373,22 @@ gpi_sim_hdl gpi_get_handle_by_index(gpi_sim_hdl base, int32_t index) {
 }
 
 gpi_iterator_hdl gpi_iterate(gpi_sim_hdl obj_hdl, gpi_iterator_sel_t type) {
+    if (obj_hdl == NULL && type == GPI_PACKAGE_SCOPES) {
+        vector<GpiImplInterface *>::iterator implIter;
+
+        LOG_DEBUG("Looking for packages over %d implementations",
+                  registered_impls.size());
+
+        for (implIter = registered_impls.begin();
+             implIter != registered_impls.end(); implIter++) {
+            GpiIterator *iter =
+                (*implIter)->iterate_handle(NULL, GPI_PACKAGE_SCOPES);
+            if (iter == NULL) continue;
+            if (!iter->empty()) return iter;
+        }
+        return NULL;
+    }
+
```

Comment:
What about the case when `obj_hdl == NULL` and `type == GPI_PACKAGE_SCOPES`? Maybe the `if` should be more inclusive and if `obj_hdl != NULL`, then an error message could be printed.